### PR TITLE
add board definitions for unphone ESP32 and ESP32S3 devices

### PR DESCRIPTION
**Pull request by hamishcunningham**
Board definitions for unphone boards: see https://iot.unphone.net/ and https://unphone.net/ for details.

### CODE REVIEW

File: boards/unphone7.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "core": "esp32",
+    "extra_flags": [
+      "-DUNPHONE_SPIN=7",
+      "-DARDUINO_FEATHER_ESP32"
+    ],
+    "f_cpu": "240000000L",
+    "f_flash": "40000000L",
+    "flash_mode": "dio",
+    "mcu": "esp32",
+    "variant": "feather_esp32"
```
Comment (valeros) [inline boards/unphone7.json:15]: Are you sure the `feather_esp32` variant is suitable for your board?
Same question for `-DARDUINO_FEATHER_ESP32`.

File: boards/unphone8.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      "-DUNPHONE_SPIN=8",
+      "-DARDUINO_USB_CDC_ON_BOOT=0",
+      "-DARDUINO_RUNNING_CORE=1",
+      "-DARDUINO_EVENT_RUNNING_CORE=1"
+    ],
+    "f_cpu": "240000000L",
+    "f_flash": "80000000L",
+    "flash_mode": "dio",
+    "mcu": "esp32s3",
+    "variant": "unphone8"
```
Comment (valeros) [inline boards/unphone8.json:17]: I'm afraid this won't work as there is no such variant in the official Arduino core for ESP32 https://github.com/espressif/arduino-esp32/tree/master/variants . Shouldn't it be added there in the first place?

File: boards/unphone8.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  },
+  "frameworks": [
+    "arduino",
+    "espidf"
+  ],
+  "name": "unPhone 8",
+  "upload": {
+    "flash_size": "8MB",
+    "maximum_ram_size": 327680,
+    "maximum_size": 4194304,
```
Comment (valeros) [inline boards/unphone8.json:33]: I guess it should be `8388608` as you specified 8MB in the `flash_size` field.

File: boards/unphone9.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    "f_flash": "80000000L",
+    "flash_mode": "dio",
+    "hwids": [
+      [
+        "0x16D0",
+        "0x1178"
+      ]
+    ],
+    "mcu": "esp32s3",
+    "variant": "unphone9"
```
Comment (valeros) [inline boards/unphone9.json:23]: Same as for `unPhone 8`. There is no such variant https://github.com/espressif/arduino-esp32/tree/master/variants

File: boards/unphone9.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  ],
+  "name": "unPhone 9",
+  "upload": {
+    "arduino": {
+      "flash_extra_images": [
+        [
+          "0x410000",
+          "variants/unphone9/tinyuf2.bin"
+        ]
+      ]
```
Comment (valeros) [inline boards/unphone9.json:43]: This won't work as `variants/unphone9/tinyuf2.bin` doesn't exist in the framework.

File: boards/unphone9.json [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    },
+    "flash_size": "8MB",
+    "maximum_ram_size": 327680,
+    "maximum_size": 4194304,
```
Comment (valeros) [inline boards/unphone9.json:47]: Same as for `unPhone 8`, should be `8388608`.

**Review by valeros — CHANGES_REQUESTED** [review]

**Comment by valeros** [comment]
Hi @hamishcunningham ! Thanks for the PR. Please see my comments above.

File: boards/unphone7.json (hunk 1 above)
Comment (hamishcunningham) [reply boards/unphone7.json:15]: In the early versions we duplicated the Huzzah32 Feather pinouts and users selected that board definition in platformio.ini

So this does work for versions before 8, when we move over to the ESP32S3 and pin mappings become a little different.

We've tested all of these platformio board defs with various test rigs, including: https://gitlab.com/hamishcunningham/unphone/-/blob/master/examples/2022/platformio.ini

File: boards/unphone8.json (hunk 2 above)
Comment (hamishcunningham) [reply boards/unphone8.json:17]: You're right! But we've submitted a PR there a few days ago:
https://github.com/espressif/arduino-esp32/pull/7058
In the meantime our fork provides that definition:
https://github.com/garethhcoleman/arduino-esp32

File: boards/unphone8.json (hunk 3 above)
Comment (hamishcunningham) [reply boards/unphone8.json:33]: Good point, will change!

File: boards/unphone9.json (hunk 4 above)
Comment (hamishcunningham) [reply boards/unphone9.json:23]: See above, a PR is pending on the core repo.

File: boards/unphone9.json (hunk 5 above)
Comment (hamishcunningham) [reply boards/unphone9.json:43]: See above, a PR is pending on the core repo.

File: boards/unphone9.json (hunk 6 above)
Comment (hamishcunningham) [reply boards/unphone9.json:47]: Tnx!

**Comment by garethhcoleman** [comment]
Just to add, the PR to add variants to arduino-esp32 is now https://github.com/espressif/arduino-esp32/pull/7109

**Comment by hamishcunningham** [comment]
Our pull request on arduino-esp32 has now been accepted
https://github.com/espressif/arduino-esp32/pull/7109
and we've checked and revised all the sizes, so hopefully this is good to go?
Tnx!

File: boards/unphone8.json [hunk 7]
```diff
@@ -0,0 +1,41 @@
+{
+  "build": {
+    "arduino":{
+      "ldscript": "esp32s3_out.ld"
```
Comment (valeros) [inline boards/unphone8.json:4]: In the PR to the Arduino core you used a special default partition. Please specify it here as well, see an [example](https://github.com/platformio/platform-espressif32/blob/develop/boards/esp32-s3-devkitc-1.json#L5).

File: boards/unphone8.json [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    },
+    "core": "esp32",
+    "extra_flags": [
+      "-DUNPHONE_SPIN=8",
+      "-DARDUINO_USB_CDC_ON_BOOT=0",
+      "-DARDUINO_RUNNING_CORE=1",
+      "-DARDUINO_EVENT_RUNNING_CORE=1"
```
Comment (valeros) [inline boards/unphone8.json:11]: There is `-DBOARD_HAS_PSRAM` in the PR to the Arduino core. Shouldn't it be added here as well for both `unphone8` and `unphone9`?

**Review by valeros — CHANGES_REQUESTED** [review]

**Comment by valeros** [comment]
Hi @hamishcunningham , thanks for the updates. Please review my comments above.

**Comment by garethhcoleman** [comment]
Thanks for your patience reviewing our submission @valeros!!

In addition to the changes you highlighted I have an additional question.

"flash_mode": "dio", is this correct? On more careful review, other boards like ESP32-S3-DevKitC-1 have it as qio, indeed I see a recent fix that changed this.

Thanks again!

**Comment by valeros** [comment]
> "flash_mode": "dio", is this correct? 

It seems you're using the QIO bootloader in the PR to the Arduino core, so it should be synchronized here as well.

**Comment by hamishcunningham** [comment]
hi @valeros 
I think 648b105405a5b04947d47873228b396cd0b6f65c addresses your comments?
Tnx!
H

**Comment by valeros** [comment]
LGTM, the PR will be merged once a new stable version of ESP32 core is released.

**Comment by garethhcoleman** [comment]
@valeros thank you, appreciate the update and glad that we have been able to address issues.

good vibes to you!

G

**Comment by hamishcunningham** [comment]
hi @valeros can this be merged now that 2.0.5 of the core is released? tnx! h

**Comment by valeros** [comment]
Thanks for the PR, merged! The boards will be available in the upcoming release of the dev-platform.

**Comment by hamishcunningham** [comment]
Thank you!

**Pull request merged**